File: solutions/Leetcode_1710/leetcode_1710.py
from typing import List


class Solution1:
    def maximum_units(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x: x[1], reverse=True)
        units = 0
        for num, unit in boxTypes:
            n = num if truckSize >= num else truckSize
            truckSize -= n
            units += n * unit
            if truckSize == 0:
                return units
        return units

File: solutions/Leetcode_1710/leetcode_1710.cc
#include <algorithm>
#include <vector>

using namespace std;

class Solution1
{
public:
    int maximumUnits(vector<vector<int>> &boxTypes, int truckSize)
    {
        sort(boxTypes.begin(), boxTypes.end(), [](vector<int> &first, vector<int> &second) {
            return first[1] > second[1];
        });
        int units = 0;
        for (auto box_type : boxTypes)
        {
            int n = min(box_type[0], truckSize);
            truckSize -= n;
            units += n * box_type[1];
            if (truckSize == 0)
                return units;
        }
        return units;
    }
};